File: leetcode/14-LongestCommonPrefix/go/longestcommonprefix.go
package main

import (
	"fmt"
	"strings"
)

func longestCommonPrefix(strs []string) string {
    if len(strs) == 0 {
        return ""
    }

    prefix := strs[0]

    for _, str := range strs {
        for strings.HasPrefix(str, prefix) == false {
            if len(prefix) == 0 {
                return ""
            }
            prefix = prefix[:len(prefix)-1]
        }
    }
    return prefix
}

func main() {
	strs := []string{"", "b"}
	fmt.Println(longestCommonPrefix(strs))
}

File: leetcode/14-LongestCommonPrefix/longestcommonprefix.cpp
#include <vector>
#include <string>

using namespace std;

class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if(strs.size() == 0) {
            return "";
        }
        string prefix = strs[0];
        for(int i = 1; i < strs.size(); i++) {
            // while occurence of substring is not present ie flower in flow until you search for flow in flow
                // prefix = slice of prefix from char 0 to prefix length -1 

            while (strs[i].find(prefix) != 0) {
                prefix = prefix.substr(0, prefix.length() - 1);
                if (prefix == "") {
                    return "";
                }
            }
        }
        return prefix;
    }
};